.github/workflows/CD.yml

```diff
@@ -7,20 +7,20 @@ on:
 concurrency: cd-${{ github.ref }}
 
 jobs:
-  build-and-deploy:
+  setup:
```

Comment:
We use the same `setup` in CI and CD workflows. What about using action composition to reduce duplication? Is this possible here?

https://github.blog/changelog/2021-08-25-github-actions-reduce-duplication-with-action-composition/

Reply:
The current setup I've made doesn't even work, it seems not likely to either. Once I get things working somehow, then I will take a look at that, thanks for the tip! Seems interesting😇

Reply:
Created a shared job for both places 🚀!